### DISCUSSION TITLE
Expose a function to return the initial state of working directory of the session

### DISCUSSION THREAD

**Discussion by kqvanity** [olimorris/persisted.nvim#127] [Ideas]
I usually have a directory containing multiple nested projects opened. I'm also reliant nvim-rooter to change my CWD and because of that plugins like Telescope would get narrow-scoped to the nested project when greping strings/files.

My initial solution was to map `<C-P>` to change CWD to parent directory when opening telescope fuzzy finder 
```lua

local reprompt_from_parent_dir = function(picker, prompt_bufnr)
  require("telescope.actions").close(prompt_bufnr)
  vim.cmd("cd ..")
  require("telescope.builtin")[picker]()
  vim.notify("Using " .. vim.fn.getcwd())
end
...
...
find_files = {
    hidden = true,
    previewer = true,
    find_command = {
         "rg", "--files", "--hidden", "--no-ignore"
    },
    mappings = {
        i = {
          ["<C-p>"] = function(...) return reprompt_from_parent_dir("find_files", ...) end
        },
    },
},
live_grep = {
    additional_args = { "--hidden", "--no-ignore" },
    mappings = {
        i = {
          ["<C-p>"] = function(...) return reprompt_from_parent_dir("live_grep", ...) end,
        },
    }
},

```

That approach works but falls short when opening a deeply nested project, and so I was wondeirng if `Persisted` could expoe a function that returns the initial session root directory so that I can use in place of `cd ..`

**Comment by olimorris** [comment]
I've just rewritten the plugin so this should be achievable with the `current` method
